src/utils/useTests.ts

```diff
@@ -1,19 +1,35 @@
+import { useGlobals } from "@storybook/manager-api";
 import { useCallback, useState } from "react";
 
-import { BrowserInfo, StoryTestFieldsFragment, TestMode } from "../gql/graphql";
+import { BrowserInfo, StoryTestFieldsFragment, TestMode, TestStatus } from "../gql/graphql";
 
 type BrowserData = Pick<BrowserInfo, "id" | "key" | "name">;
 type ModeData = Pick<TestMode, "name">;
 
+const useGlobalValue = (key: string) => {
+  try {
+    return useGlobals()[0][key];
+  } catch (e) {
+    return null;
+  }
+};
+
 /**
  * Pick a single test+comparison (by viewport+browser) from a set of tests
  * for the same story.
  */
 export function useTests(tests: StoryTestFieldsFragment[]) {
-  const [selectedBrowserId, onSelectBrowserId] = useState<BrowserData["id"]>(
-    tests[0]?.comparisons[0]?.browser.id
-  );
-  const [selectedModeName, onSelectModeName] = useState<ModeData["name"]>(tests[0]?.mode.name);
+  const theme = useGlobalValue("theme");
+  const [selectedBrowserId, onSelectBrowserId] = useState<BrowserData["id"]>(() => {
+    const test = tests.find(({ status }) => status !== TestStatus.Passed) || tests[0];
+    return test?.comparisons[0]?.browser.id;
+  });
+  const [selectedModeName, onSelectModeName] = useState<ModeData["name"]>(() => {
+    const changed = tests.filter(({ status }) => status !== TestStatus.Passed);
```

Comment:
QQ: Are these tests limited to `testsForStory`?

Reply:
Yes, but `testsForStory` is filtered by `storyId` and returns max 100 results. It's unlikely that anyone would have that many tests for one story though. That's only possible if you have >100 modes/viewports. Seems unlikely.